src/PhotonBackground.cpp

```diff
@@ -122,52 +112,61 @@ void PhotonField::initRedshiftScaling() {
-			throw std::runtime_error("PhotonField::checkInputData: length of photon energy input is unequal to length of photon density input");
+			throw std::runtime_error("TabularPhotonField::checkInputData: length of photon energy input is unequal to length of photon density input");
 	}
 
 	for (int i = 0; i < this->photonEnergies.size(); ++i) {
 		double ePrevious = 0.;
 		double e = this->photonEnergies[i];
 		if (e <= 0.)
-			throw std::runtime_error("PhotonField::checkInputData: a value in the photon energy input is not positive");
+			throw std::runtime_error("TabularPhotonField::checkInputData: a value in the photon energy input is not positive");
 		if (e <= ePrevious)
-			throw std::runtime_error("PhotonField::checkInputData: photon energy values are not strictly increasing");
+			throw std::runtime_error("TabularPhotonField::checkInputData: photon energy values are not strictly increasing");
 		ePrevious = e;
 	}
 
 	for (int i = 0; i < this->photonDensity.size(); ++i) {
 		if (this->photonDensity[i] < 0.)
-			throw std::runtime_error("PhotonField::checkInputData: a value in the photon density input is negative");
+			throw std::runtime_error("TabularPhotonField::checkInputData: a value in the photon density input is negative");
 	}
 
-	if (this->hasRedshiftDependence) {
+	if (this->isRedshiftDependent) {
 		if (this->redshifts[0] != 0.)
-			throw std::runtime_error("PhotonField::checkInputData: redshift input must start with zero");
+			throw std::runtime_error("TabularPhotonField::checkInputData: redshift input must start with zero");
 
 		for (int i = 0; i < this->redshifts.size(); ++i) {
 			double zPrevious = -1.;
 			double z = this->redshifts[i];
 			if (z < 0.)
-				throw std::runtime_error("PhotonField::checkInputData: a value in the redshift input is negative");
+				throw std::runtime_error("TabularPhotonField::checkInputData: a value in the redshift input is negative");
 			if (z <= zPrevious)
-				throw std::runtime_error("PhotonField::checkInputData: redshift values are not strictly increasing");
+				throw std::runtime_error("TabularPhotonField::checkInputData: redshift values are not strictly increasing");
 			zPrevious = z;
 		}
 
 		for (int i = 0; i < this->redshiftScalings.size(); ++i) {
 			double scal = this->redshiftScalings[i];
 			if (scal <= 0.)
-				throw std::runtime_error("PhotonField::checkInputData: a value in the redshift input for scaling is not positive");
+				throw std::runtime_error("TabularPhotonField::checkInputData: a value in the redshift input for scaling is not positive");
 		}
 	}
 }
 
+BlackbodyPhotonField::BlackbodyPhotonField(std::string fieldName, double blackbodyTemperature) {
+	this->fieldName = fieldName;
+	this->blackbodyTemperature = blackbodyTemperature;
+}
+
+double BlackbodyPhotonField::getPhotonDensity(double ePhoton) const {
+	return 8 * M_PI * std::pow(ePhoton / (h_planck * c_light), 3) / (std::exp(ePhoton / (k_boltzmann * this->blackbodyTemperature)) - 1);
```

Comment:
you should use pow_integer<3>( ... ) instead of std::pow as this (see Common.h) is 10x faster than std:pow for integers. Even a loop is up to 2x faster for integers (if the exponent is not known at compile time).

std::exp( X ) -1 should be replaced by std::expm1(X) due to precision